### DISCUSSION TITLE
Vectorization process?

### DISCUSSION THREAD

**Discussion by tondeaf** [brianpetro/obsidian-smart-connections#249] [General]
Ok, I just found this gem as I was thinking, "How do I vectorize my obsidian DB and then query it?" Et voila! 

So, a few questions.

1. How does it generate the vectors? Are the notes uploaded to openAI and returned as vectors? It doesn't *look* like it, because the number is just skyrocketing higher and higher...so it looks like it is vectorizing "locally". Is that true? 

2. How long does it take to vectorize? Per word or token? I have 3 million words, so....? 

For 1+2, does it cost money to vectorize? How much? :o 

4. What is the number that cycles upward when you turn it on? Does that just go on forever? I tried it with a tiny DB and it's at 20k+? Why? Do I need to restart?

5. If I have multiple VAULTS, will they all be vectorized at once? 

6. Where will the vector results be stored?

**Comment by nomadphase** [comment]
As I understand it, it uses 

1 The plugin integrates [OpenAI Embeddings](https://beta.openai.com/docs/guides/embeddings), a technology from the organization behind ChatGPT, to use AI that finds connections between notes. Instead of matching keywords, the AI interprets your notes as 1,536-dimension vectors

2 the vectorisation shouldn't take too long. I have a larger vault and it was pretty quick

3 the vector results are stored in a json file in a hidden folder in your obsidian vault folder
